GVRf/Framework/framework/src/main/jni/engine/renderer/renderer.cpp

```diff
@@ -161,17 +375,115 @@ void Renderer::cullFromCamera(Scene *scene, Camera* camera,
     // 2. Iteratively execute frustum culling for each root object (as well as its children objects recursively)
     SceneObject *object = scene->getRoot();
     if (DEBUG_RENDERER) {
-        LOGD("FRUSTUM: start frustum culling for root %s\n", object->name().c_str());
+        LOGD("FRUSTUM: start frustum culling for root %s\n",
+                object->name().c_str());
     }
-    frustum_cull(camera->owner_object()->transform()->position(), object, frustum, scene_objects, scene->get_frustum_culling(), 0);
+    frustum_cull(camera->owner_object()->transform()->position(), object,
+            frustum, scene_objects, scene->get_frustum_culling(), 0);
     if (DEBUG_RENDERER) {
-        LOGD("FRUSTUM: end frustum culling for root %s\n", object->name().c_str());
+        LOGD("FRUSTUM: end frustum culling for root %s\n",
+                object->name().c_str());
     }
     // 3. do occlusion culling, if enabled
     occlusion_cull(scene, scene_objects, shader_manager, vp_matrix);
 }
 
+/**
+ * Set the render states for render data
+ */
+void Renderer::setRenderStates(RenderData* render_data, RenderState& rstate) {
+
+    if (!(rstate.render_mask & render_data->render_mask()))
+        return;
+
+    if (render_data->offset()) {
+        GL(glEnable (GL_POLYGON_OFFSET_FILL));
+        GL(
+                glPolygonOffset(render_data->offset_factor(),
+                        render_data->offset_units()));
+    }
+    if (!render_data->depth_test()) {
```

Comment:
Do you need to have an else clause to enable Depth test, blend, etc or is the assumption that the render state will have these states set?